src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx

```diff
@@ -239,73 +242,93 @@ const ReleaseDataUploadsSection = ({
               </>
             )}
 
-            {dataFiles.map(dataFile => (
-              <SummaryList
-                key={dataFile.filename}
-                additionalClassName="govuk-!-margin-bottom-9"
-              >
-                <SummaryListItem term="Subject title">
-                  <h4 className="govuk-heading-m">{dataFile.title}</h4>
-                </SummaryListItem>
-                <SummaryListItem term="Data file">
-                  <ButtonText
-                    onClick={() =>
-                      service
-                        .downloadDataFile(releaseId, dataFile.filename)
-                        .catch(handleApiErrors)
-                    }
-                  >
-                    {dataFile.filename}
-                  </ButtonText>
-                </SummaryListItem>
-                <SummaryListItem term="Metadata file">
-                  <ButtonText
-                    onClick={() =>
-                      service
-                        .downloadDataMetadataFile(
-                          releaseId,
-                          dataFile.metadataFilename,
-                        )
-                        .catch(handleApiErrors)
-                    }
+            <Accordion id="data-files">
+              {dataFiles.map((dataFile, index) => (
+                <AccordionSection
+                  /* eslint-disable-next-line react/no-array-index-key */
+                  key={`${dataFile.title}-${index}`}
+                  headingId={`${dataFile.title}-${index}`}
+                  heading={`${dataFile.title}`}
+                  onToggle={() => {
+                    const accId = `${dataFile.title}-${index}`;
+                    const opened = openedAccordions;
+                    // eslint-disable-next-line no-unused-expressions
+                    opened.includes(accId)
+                      ? opened.splice(opened.indexOf(accId), 1)
```

Comment:
This mutates the the state directly and is a no-no in React as it can cause weird bugs (although in this case it probably won't matter too much).

Regardless, we should probably rewrite this in an immutable fashion. There's a few ways you could do this, but Lodash's `remove` function is probably the easiest.